// @typegoose/typegoose/lib/internal/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlreadyMerged = exports.Severity = exports.PropType = exports.DecoratorKeys = void 0;
/**
 * Collection of Reflect Types for easy maintenance
 */
var DecoratorKeys;
(function (DecoratorKeys) {
    /** Get the Typescript assigned Type at runtime */
    DecoratorKeys["Type"] = "design:type";
    /**
     * "@prop" Cache
     * -> Use only for a class
     */
    DecoratorKeys["PropCache"] = "typegoose:properties";
    /**
     * Storage location for Model Options
     * -> Use only for a class
     */
    DecoratorKeys["ModelOptions"] = "typegoose:options";
    /**
     * Storage location for Indexes
     * -> Use only for a class
     */
    DecoratorKeys["Index"] = "typegoose:indexes";
    /**
     * Storage location for Search Indexes
     * -> Use only for a class
     */
    DecoratorKeys["SearchIndex"] = "typegoose:searchIndexes";
    /**
     * Storage location for Plugins
     * -> Use only for a class
     */
    DecoratorKeys["Plugins"] = "typegoose:plugins";
    /**
     * Storage location for Pre-Hooks
     * -> Use only for a class
     */
    DecoratorKeys["HooksPre"] = "typegoose:hooksPre";
    /**
     * Storage location for Post-Hooks
     * -> Use only for a class
     */
    DecoratorKeys["HooksPost"] = "typegoose:hooksPost";
    /**
     * Storage location for Virtual Populates
     * -> Use only for a class
     */
    DecoratorKeys["VirtualPopulate"] = "typegoose:virtualPopulate";
    /**
     * Storage location for Query Methods
     * -> Use only for a class
     */
    DecoratorKeys["QueryMethod"] = "typegoose:queryMethod";
    /**
     * Storage location for Nested Discriminators
     * -> Use only for a class
     */
    DecoratorKeys["NestedDiscriminators"] = "typegoose:nestedDiscriminators";
    /**
     * Storage location for Cached Schemas
     * -> Use only for a class
     */
    DecoratorKeys["CachedSchema"] = "typegoose:cachedSchema";
})(DecoratorKeys || (exports.DecoratorKeys = DecoratorKeys = {}));
/** This Enum is meant for baseProp to decide for different props (like if it is an arrayProp or prop or mapProp) */
var PropType;
(function (PropType) {
    PropType[PropType["ARRAY"] = 0] = "ARRAY";
    PropType[PropType["MAP"] = 1] = "MAP";
    PropType[PropType["NONE"] = 2] = "NONE";
})(PropType || (exports.PropType = PropType = {}));
/** Severity levels for soft-warnings */
var Severity;
(function (Severity) {
    Severity[Severity["ALLOW"] = 0] = "ALLOW";
    Severity[Severity["WARN"] = 1] = "WARN";
    Severity[Severity["ERROR"] = 2] = "ERROR";
})(Severity || (exports.Severity = Severity = {}));
/**
 * Symbol to track if options have already been merged
 * This is to reduce the "merge*" calls, which dont need to be run often if already done
 */
exports.AlreadyMerged = Symbol('MOAlreadyMergedOptions');
